pkg/process/metadata/parser/service.go

```diff
@@ -334,3 +336,22 @@ func parseCommandContextJava(args []string) string {
 
 	return "java"
 }
+
+// parseCommandContextDotnet extracts metadata from a dotnet launcher command line
+func parseCommandContextDotnet(args []string) string {
```

Comment:
is that intended to run on linux as well?
if so, the `.dll` is still the expected suffix?

Reply:
yes I also found this a bit weird (I expected `.so` or others). .NET is cross platform but the build output are `.dll` and they are PE executables. Then the interpreter will adapt to the arch and the os

To double check I build some samples on my mac :
```
dotnet build
...
find . | grep dll
...
./logger-message-generator/bin/Debug/net7.0/logger-message-generator.dll
...
file ./logger-message-generator/bin/Debug/net7.0/logger-message-generator.dll
./logger-message-generator/bin/Debug/net7.0/logger-message-generator.dll: PE32 executable (console) Intel 80386 Mono/.Net assembly, for MS Windows

And then
dotnet ./logger-message-generator/bin/Debug/net7.0/logger-message-generator.dll
....
{
  "EventId": 23,
  "LogLevel": "Information",
 .....
  }
}

so yes it's expected this way

Reply:
Had the same thought and seeing that output is 🤯 to me